lib/Saml2/Auth.php

```diff
@@ -113,13 +120,80 @@ public function processResponse($requestId = null)
         } else {
             $this->_errors[] = 'invalid_binding';
             throw new OneLogin_Saml2_Error(
-                'SAML Response not found, Only supported HTTP_POST Binding',
+                'SAML Response not found, Only supporting HTTP-POST and HTTP-Artifact Bindings',
                 OneLogin_Saml2_Error::SAML_RESPONSE_NOT_FOUND
             );
         }
     }
 
     /**
+     * Performs artifact resolution
+     * @param string $encodedArtifact
+     * @return OneLogin_Saml2_Response
+     * @throws OneLogin_Saml2_Error
+     */
+    protected function _resolveArtifact($encodedArtifact) {
+        if (!($destination = $this->getARSurl())) {
+            throw new OneLogin_Saml2_Error(
+                'Missing artifact resolution service URL.',
+                OneLogin_Saml2_Error::SETTINGS_INVALID
+            );
+        }
+
+        $idpData = $this->_settings->getIdPData();
+        $artifact = base64_decode($encodedArtifact);
+        $sourceId = bin2hex(substr($artifact, 4, 20));
+
+        if ($sourceId !== sha1($idpData['entityId'])) {
+            throw new OneLogin_Saml2_Error(
+                'Invalid IdP Source',
+                OneLogin_Saml2_Error::SAML_ARS_SOURCE_INVALID
+            );
+        }
+
+        $request = new OneLogin_Saml2_ArtifactResolve($this->_settings, $destination, $encodedArtifact);
+        $rootXml = $request->getXML(true);
+
+        $xml = <<<ARS
+<?xml version="1.0"?>
+<soap-env:Envelope xmlns:soap-env="http://schemas.xmlsoap.org/soap/envelope/"><soap-env:Header/>
+    <soap-env:Body>$rootXml</soap-env:Body>
+</soap-env:Envelope>
+ARS;
+
+        // POST the data using cURL
+        // TODO support basic auth
+        $ch = curl_init();
```

Comment:
I consider calling curl directly a bad practice.
As php 5.3 support is required I would recommend using Guzzle 3 https://github.com/guzzle/guzzle3.
Why use curl when php has its own soap client http://php.net/manual/en/class.soapclient.php?